crates/edr_napi/test/provider.ts

```diff
@@ -415,3 +515,48 @@ function assertEqualMemory(stepMemory: Buffer | undefined, expected: Buffer) {
 
   assert.isTrue(stepMemory.equals(expected));
 }
+
+function assertJsonRpcFormatNormalised(trace: RpcDebugTraceResult) {
+  assert.isBoolean(trace.failed);
+  assert.typeOf(trace.gas, "number");
+  assert.isString(trace.returnValue);
+  assert.isArray(trace.structLogs);
+
+  trace.structLogs.forEach((log: RpcDebugTraceLogItem) => {
+    assert.typeOf(log.pc, "number");
+    assert.typeOf(log.op, "string");
+    assert.typeOf(log.gas, "number");
+    assert.typeOf(log.gasCost, "number");
+    assert.typeOf(log.depth, "number");
+    assert.typeOf(log.memSize, "number");
+
+    if ("stack" in log) {
+      assert.isArray(log.stack);
+      log.stack?.forEach((item) => {
+        assert.isString(item);
+        // assert.isFalse(item.startsWith("0x"));
+      });
+    }
+
+    if ("memory" in log) {
+      assert.isArray(log.memory);
+      log.memory?.forEach((item) => {
+        assert.isString(item);
+      });
+    }
+
+    if ("storage" in log) {
+      assert.isObject(log.storage);
+      Object.entries(log.storage!).forEach(([key, value]) => {
+        assert.isString(key);
+        assert.isString(value);
+        // assert.isFalse(key.startsWith("0x"));
+        // assert.isFalse(value.startsWith("0x"));
```

Comment:
Why are these two disabled?

Reply:
I disabled these two because at this moment, the normalisation is still being done in Hardhat so it is still correct to have `0x` at the start of the strings.
Here is the reference in Hardhat: https://github.com/NomicFoundation/hardhat/blob/024d72b09c6edefb00c012e9514a0948c255d0ab/v-next/hardhat/src/internal/builtin-plugins/network-manager/edr/utils/convert-to-edr.ts#L197-L203

The Hardhat tests also operate on this assumption so I believe these two asserts should be disabled until the normalisation is completely removed from Hardhat.

Reply:
Ah perfect. Thank you